api/v0/runtime.proto

```diff
@@ -28,168 +28,130 @@
+  rpc Stop(CellStopRequest) returns (CellStopResponse) {}
 }
 
-// Executable is the lowest level of compute that Aurae can execute. A basic process.
+/// The most primitive workload in Aurae, a standard executable process.
 message Executable {
-  meta.AuraeMeta meta = 1;
-
-  /// Command resembles systemd's ExecStart. This is the shell command (with arguments) you intend to execute.
-  string command = 3;
-
-  /// Comment is an arbitrary (user defined) comment used to identify the Executable at runtime.
-  string comment = 4;
-}
-
-// ExecutableStatus is only returned after a process completes. Because runtime is a synchronous subsystem
-// this will only return upon a terminated process.
-message ExecutableStatus {
-  meta.AuraeMeta meta = 1;
-  meta.ProcessMeta proc = 2;
-  meta.Status status = 3;
-
-  // The full stdout data.
-  string stdout = 4;
-
-  // The full stderr data.
-  string stderr = 5;
-
-  // The exit code (return code) of the process after termination.
-  string exit_code = 6;
-}
-
-// Pod is a group of containers running in a spawned Aurae instance.
-message Pod {
-  meta.AuraeMeta meta = 1;
-
-  // The cell to create within the spawned Aurae instance.
-  Cell cell = 2;
-}
-
-// PodStatus is the status of a completed pod and its subsequent containers.
-message PodStatus {
-  meta.AuraeMeta meta = 1;
-  meta.Status status = 3;
-
-  // The response of the necessary spawn for a pod.
-  SpawnResponse spawn_response = 4;
-
-  // If the spawn is successful, the nested Aurae instance.
-  Instance instance = 5;
-
-  // If the cell creation is successful, the cell of containers running in the spawned Aurae instance.
-  CellStatus cell_status = 6;
-
+  string name = 1;
+  string command = 2;
+  string description = 3;
+  string cell_name = 4;
 }
 
-// Container represents is an OCI compliant container image which can be executed.
-message Container {
-  meta.AuraeMeta meta = 1;
-
-  // OCI compliant image.
-  string image = 2;
+/// A reference to an executable.
+message ExecutableReference {
+  string cell_name = 1;
+  string executable_name = 2;
 }
 
-// ContainerStatus is the status of a container after it has been executed.
-message ContainerStatus {
-  meta.AuraeMeta meta = 1;
-  meta.ProcessMeta proc = 2;
-  meta.Status status = 3;
+/// An isolation resource used to divide a system into smaller resource boundaries.
+message Cell {
+  string name = 1;
+  ControlGroupOptions cgroup = 2;
+  NamespaceShare share = 3;
 }
 
-// A long lived virtual machine which will persist on termination.
-message VirtualMachine {
-  meta.AuraeMeta meta = 1;
-  string name = 2;
-  string image = 3;
+/// Resource parameters for control groups (cgroups)
+/// Build on the [cgroups-rs](https://github.com/kata-containers/cgroups-rs) crate.
+/// See [examples](https://github.com/kata-containers/cgroups-rs/blob/main/tests/builder.rs)
+message ControlGroupOptions {
+  string name = 1;
+  string cpus = 2;
```

Comment:
Why is `cpus` a `string`? Is this like how k8s can have multiple suffixes? If so, Protocol buffers has better typing that can capture intent. Something like:

```proto
message Cpu {
    oneof unit {
        u32 percent = 1;
        u32 millicores = 2;
    }
}
```

Reply:
This is just to map the library for now, we will expose controls as needed once the cells are initiated